Lib/test/datetimetester.py

```diff
@@ -3897,6 +3927,19 @@ def test_repr(self):
         self.assertEqual(repr(self.theclass(23, 15, 0, 0)),
                          "%s(23, 15)" % name)
 
+    def test_repr_subclass(self):
```

Comment:
Let's add a docstring describing what this test is asserting.

```suggestion
    def test_repr_subclass(self):
        """Subclasses should have bare names in the repr (gh-107773)."""
```